src/Optimize/VariableSet.h

```diff
@@ -124,6 +124,18 @@ struct VariableSet
     return -1;
   }
 
+  inline int getLoc(const std::string& vname) const
```

Comment:
I call this in resetParameters for both the CountingJastrow and GaussianFunctor classes. I add non-optimizable parameters to their VariableSet so that they their values may be read by the opt.xml updating function for reasons I describe in the next comment. 

Values in the Index array are updated before the non-optimizable parameters are removed, yet are not updated after. As a result, using getIndex returns the wrong index, and the incorrect parameter is updated.

Alternatives here are to reset the indices after these parameters are removed, either directly in the VariableSet call, or by adding another call to checkOutParameters either in QMCCostFunctionBase or to each of the wavefunction's resetParameters functions. 

I'm sort of puzzled why this indexing scheme is built on these locally-cached indices when it produces this additional complexity.